File: src/tag.js
//Provides: caml_binaryen_add_tag
//Requires: caml_jsstring_of_string
function caml_binaryen_add_tag(wasm_mod, name, params, results) {
  return wasm_mod.addTag(
    caml_jsstring_of_string(name),
    params,
    results,
  );
}

//Provides: caml_binaryen_get_tag
//Requires: caml_jsstring_of_string
function caml_binaryen_get_tag(wasm_mod, name) {
  return wasm_mod.getTag(caml_jsstring_of_string(name));
}

//Provides: caml_binaryen_remove_tag
//Requires: caml_jsstring_of_string
function caml_binaryen_remove_tag(wasm_mod, name) {
  return wasm_mod.removeTag(caml_jsstring_of_string(name));
}

//Provides: caml_binaryen_tag_get_name
//Requires: Binaryen
//Requires: caml_string_of_jsstring
function caml_binaryen_tag_get_name(tag) {
  var tag_info = Binaryen.getTagInfo(tag);
  return caml_string_of_jsstring(tag_info.name);
}

//Provides: caml_binaryen_tag_get_params
//Requires: Binaryen
function caml_binaryen_tag_get_params(tag) {
  var tag_info = Binaryen.getTagInfo(tag);
  return tag_info.params;
}

//Provides: caml_binaryen_tag_get_results
//Requires: Binaryen
function caml_binaryen_tag_get_results(tag) {
  var tag_info = Binaryen.getTagInfo(tag);
  return tag_info.results;
}

File: src/tag.ml
type t

external add_tag : Module.t -> string -> Type.t -> Type.t -> t
  = "caml_binaryen_add_tag"
(** Module, name, params, result *)

external get_tag : Module.t -> string -> t = "caml_binaryen_get_tag"
external remove_tag : Module.t -> string -> unit = "caml_binaryen_remove_tag"
external get_name : t -> string = "caml_binaryen_tag_get_name"
external get_params : t -> Type.t = "caml_binaryen_tag_get_params"
external get_results : t -> Type.t = "caml_binaryen_tag_get_results"
